functions/src/functions/users.ts

```diff
@@ -0,0 +1,293 @@
+//
+// This source file is part of the ENGAGE-HF project based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import admin from 'firebase-admin'
+import { https } from 'firebase-functions'
+import { type CallableRequest, onCall } from 'firebase-functions/v2/https'
+import { type Result } from './types.js'
+import { type UserAuthenticationInformation } from '../models/invitation.js'
+import {
+  type Admin,
+  type Clinician,
+  type Patient,
+  type User,
+} from '../models/user.js'
+import { FirestoreService } from '../services/database/firestoreService.js'
+import { SecurityService } from '../services/securityService.js'
+
+export interface GetUsersInformationInput {
+  includeClinicianData?: boolean
+  includePatientData?: boolean
+  includeUserData?: boolean
+  userIds?: string[]
+}
+
+export interface UserInformation {
+  auth: UserAuthenticationInformation
+  clinician?: Clinician
+  patient?: Patient
+  user?: User
+}
+
+export type GetUsersInformationOutput = Record<string, Result<UserInformation>>
+
+export const getUsersInformationFunction = onCall(
```

Comment:
Not part of this PR, but a future issue should have a service account attached to these functions to ensure that they are all executed in a minimally scoped service account, e.g, how we do it in PAWS: https://github.com/StanfordBDHG/PediatricAppleWatchStudy/blob/main/functions/index.js#L20